src/App.tsx

```diff
@@ -8,26 +8,27 @@ const projectRegex = /^P[a-zA-Z0-9]{27}$/;
 const App = () => {
 	const [error, setError] = useState(false);
 
-	const baseUrl = process.env.REACT_APP_DESCOPE_BASE_URL || '';
+	const baseUrl =
+		process.env.REACT_APP_DESCOPE_BASE_URL || window.location.origin;
 
 	let projectId = '';
 
 	// first, take project id from env
-	const envProjectId = process.env.DESCOPE_PROJECT_ID;
-	if (envProjectId && projectRegex.test(envProjectId)) {
-		projectId = envProjectId;
-	}
+	const envProjectId = projectRegex.exec(
+		process.env.DESCOPE_PROJECT_ID ?? ''
+	)?.[0];
 
-	// If exists in URI we will take it from the URI
+	// If exists in URI we will take it from the URI and save it in local storage
 	const pathnameProjectId = window.location.pathname?.split('/').at(-1) || '';
-	if (pathnameProjectId) {
-		if (projectRegex.test(pathnameProjectId)) {
-			projectId = pathnameProjectId;
-		} else {
-			console.log(`Invalid Project ID: ${pathnameProjectId}`); // eslint-disable-line
-		}
+	if (pathnameProjectId && projectRegex.test(pathnameProjectId)) {
+		projectId = pathnameProjectId;
```

Comment:
this line is not needed as you override it in the next block.. 
better to keep using `pathnameProjectId` so it will be more clear

Reply:
it's not, at the end of this block the user switches pages and the page rerenders, I'll make a comment so it's clearer